common/rpc-directory/mock-redis.mjs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+class MockRedis {
+    /**
+     * Create the Mock Redis.
+     * @constructor
+     */
+    constructor() {
+        this._data = new Map();
+        this.clients = [];
+        this.callbacks = [];
+        this.subscribed = {};
+    }
+
+    createClient() {
+        let newClient = new MockRedisClient(this);
+        this.clients.push(newClient);
+        return newClient;
+    }
+}
+
+class MockRedisClient{
+    constructor(mockRedis) {
+        this._mockRedis = mockRedis;
+        this._data = mockRedis._data;
+    }
+
+    get(key, cb) {
+        if (this._data.has(key)){
+            cb(null, this._data.get(key));
+        } else {
+            cb(null, null); // If the key does not exist the special value nil is returned.
+        }
+    }
+
+    set(key, value, cb) {
```

Comment:
We currently use the form set([key, value], cb).

Reply:
and we need to support 'NX' attribute.

Reply:
Resolved.